### PR TITLE
Upgrade controller-runtime to v0.3.0

### PR DESCRIPTION
**Pull request by kron4eg**
* Upstream k8s libs are updated to kubernetes-1.15.4 release tag
* Fixed many API breaking changes
* Logic regarding checking version of CCM is removed

**What this PR does / why we need it**:
Takes advantage of newer and better controller-runtime APIs

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Upgrade controller-runtime to v0.3.0
```

### CODE REVIEW

**Comment by kron4eg** [comment]
/hold

hack with cache drop (resolved upstream) to be removed

File: pkg/templates/nodelocaldns/nodelocaldns.go [hunk 1]
```diff
@@ -98,6 +98,8 @@ func Deploy(s *state.State) error {
 		return errors.New("kubernetes client not initialized")
 	}
 
+	s.Logger.Infoln("Ensure node local DNS…")
```
Comment (xmudrii) [inline pkg/templates/nodelocaldns/nodelocaldns.go]: I think the feature is called NodeLocalDNSCache. I'm not sure how exactly it is written, but I think there is cache in the name.

File: pkg/certificate/pki.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Following code is a slightly modified copy from
+// k8s.io/kubernetes/cmd/kubeadm/app/util/pkiutil
```
[the remark this answers is not in the document]
Comment (kron4eg) [reply pkg/certificate/pki.go]: @xmudrii please provide more text that you may have in mind.

[the remark this answers is not in the document]
Comment (xmudrii) [reply pkg/certificate/pki.go]: @kron4eg I was thinking about writing down what's exactly modified or what is the reason why we modified the package and why we don't import the `kubeadm` package.

[the remark this answers is not in the document]
Comment (kron4eg) [reply pkg/certificate/pki.go]: >what's exactly modified

means — only few required methods are taken from upstream, not whole package

[the remark this answers is not in the document]
Comment (kron4eg) [reply pkg/certificate/pki.go]: OK, updated

File: .wwhrd.yml [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 whitelist:
-  - Apache-2.0
-  - MIT
-  - NewBSD
-  - FreeBSD
-  - LGPL-3.0
-  - MPL-2.0
+- Apache-2.0
+- FreeBSD
+- ISC
```
[the remark this answers is not in the document]
Comment (kdomanski) [reply .wwhrd.yml]: Affirmative.

**Pull request merged**